### PR TITLE
feat: unmanage edition

### PR DESCRIPTION
**Pull request by acpana**
### Change description

Provide support to unmanage the spec.settings.edition field.

#### Does this PR add something which needs to be 'release noted'?

```release-note

```

- [ ] Reviewer reviewed release note.

#### Additional documentation e.g., references, usage docs, etc.:

```docs

```

#### Intended Milestone

Please indicate the intended milestone. 
- [ ] Reviewer tagged PR with the actual milestone.

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

**Comment by acpana** [comment]
PR might've had the "mockkubiapiserver doesn't support annotations" statement but I need to investigate that further

**Review by cheftako — COMMENTED** [review]
This is a great feature to add! Thanks for your contribution. The implementation looks solid and follows existing patterns in the codebase, which is excellent for consistency. The e2e test is also well-designed and covers the new functionality thoroughly.

I have a few minor suggestions for improvement:

**Clarity and Consistency:**

1.  **Annotation Name:** The annotation used in the code is `cnrm.cloud.google.com/unmanaged`, but the commit message and the established pattern in other controllers (like BigQueryTable) use `cnrm.cloud.google.com/unmanaged-fields`. Could you please clarify which one is intended and ensure consistency? Using `unmanaged-fields` would be more consistent with the rest of the codebase.

2.  **Error Message:** In `pkg/controller/direct/sql/sqlinstance_controller.go`, the error message "unmanaging field `%s` is not supported" is helpful. It could be even more helpful if it also listed the currently supported fields for unmanagement. For example: "unmanaging field `%s` is not supported, supported fields are: 'spec.settings.edition'".

**Pull Request Process:**

1.  **Release Note:** This pull request introduces a new feature. Please add a release note to the pull request description so that users are aware of this new capability.

2.  **Milestone:** Please fill out the "Intended Milestone" section in the pull request description.

Once these minor points are addressed, this PR will be in great shape. Thanks again for the excellent work!

File: pkg/controller/direct/sql/sqlinstance_controller.go [hunk 1]
```diff
@@ -93,14 +97,26 @@ func (m *sqlInstanceModel) AdapterForObject(ctx context.Context, kube client.Rea
...
-	}, nil
+	}
+
+	unmanaged, ok := obj.GetAnnotations()["cnrm.cloud.google.com/unmanaged"]
+	if ok && unmanaged != "" {
+		unmanagedFields := strings.Split(unmanaged, ",")
+		for _, field := range unmanagedFields {
+			if field != unmanageEditionAnnotationValue {
+				return nil, fmt.Errorf("unmanaging field `%s` is not supported, supported fields are: %s", field, unmanageEditionAnnotationValue)
+			}
```
Comment (cheftako) [inline pkg/controller/direct/sql/sqlinstance_controller.go:115]: Would be nice to dedupe. Otherwise we should add a test with a duplicate to ensure it doesn't cause problems.

File: tests/e2e/testdata/scenarios/sql/unmanage-edition/script.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    edition: ENTERPRISE_PLUS
+  resourceID: sqlinstance-${uniqueId}
+---
+# 02: Add the unmanaged annotation and acquire the resource. No change expected.
+apiVersion: sql.cnrm.cloud.google.com/v1beta1
+kind: SQLInstance
+metadata:
+  name: sqlinstance-${uniqueId}
+  annotations:
+    cnrm.cloud.google.com/unmanaged: "spec.settings.edition,spec.settings.edition"
```
Comment (acpana) [inline tests/e2e/testdata/scenarios/sql/unmanage-edition/script.yaml:50]: Will follow up for better code handling (dedup, etc.)

For now, this test shows that repeated values do not break the extraction logic.

File: pkg/controller/direct/sql/sqlinstance_controller.go (hunk 1 above)
Comment (cheftako) [reply pkg/controller/direct/sql/sqlinstance_controller.go:115]: I see you added a "spec.settings.edition,spec.settings.edition" resolving for now.

File: pkg/controller/direct/sql/sqlinstance_controller.go [hunk 3]
```diff
@@ -292,39 +316,46 @@ func (a *sqlInstanceAdapter) Update(ctx context.Context, updateOp *directbase.Up
...
+				return err
+			}
+
+			updated, err := a.sqlInstancesClient.Get(a.projectID, a.resourceID).Context(ctx).Do()
+			if err != nil {
+				return fmt.Errorf("getting SQLInstance %s failed: %w", a.resourceID, err)
+			}
 
-		log.V(2).Info("instance edition updated", "op", op, "instance", updated)
+			log.Info("instance edition updated", "op", op, "instance", updated)
```
Comment (cheftako) [inline pkg/controller/direct/sql/sqlinstance_controller.go]: Not blocking but why drop the V(2)?

File: pkg/controller/direct/sql/sqlinstance_defaults.go [hunk 4]
```diff
@@ -61,10 +61,21 @@ func ApplySQLInstanceGCPDefaults(in *krm.SQLInstance, out *api.DatabaseInstance,
 		// GCP default DiskType is PD_SSD.
 		out.Settings.DataDiskType = "PD_SSD"
 	}
-	if in.Spec.Settings.Edition == nil {
-		// GCP default Edition is ENTERPRISE.
+	isEditionUnmanaged := false
+	for _, field := range unmanagedFields {
+		if field == "spec.settings.edition" {
+			isEditionUnmanaged = true
+			break
```
Comment (cheftako) [inline pkg/controller/direct/sql/sqlinstance_defaults.go:68]: Fine for now but break will become a problem when we want to support more unmanaged field paths.

**Review by cheftako — APPROVED** [review]
/lgtm
